### PR TITLE
enhancement(external docs): VRL docs fine-tuning

### PR DESCRIPTION
**Pull request by lucperkins**
This PR does a once-over on the VRL docs, offering some spelling fixes, some nitpicky things, and some rewordings.

### CODE REVIEW

File: docs/reference/remap/concepts/function.cue [hunk 1]
```diff
@@ -1,18 +1,20 @@
...
 		"""
 
 	characteristics: {
 		fallibility: {
 			title:       "Fallibility"
 			description: """
-				Of particular note for functions is fallibility. [Fail-safety](\(urls.vrl_fail_safety)) is a defining
-				characteristic of VRL and function fallibility is core to achieving that feature.
+				Some VRL functions are *fallible*, meaning that they can error. Any potential errors thrown by fallible
+				functions must be handled, a requirements enforced at compile time.
```
Comment (binarylogic) [inline docs/reference/remap/concepts/function.cue:14]:
```suggestion
				functions must be handled, a requirement enforced at compile time.
```

File: docs/reference/remap/concepts/literal.cue [hunk 2]
```diff
@@ -1,7 +1,7 @@
 remap: concepts: literal: {
 	title:       "Literal"
 	description: """
-		Like other languages, [literals](\(urls.vrl_literals)) in VRL are values written exactly as they are meant to
-		be interpreted.
+		As in most other languages, [literals](\(urls.vrl_literals)) in VRL are values written exactly as they are meant
+		to be interpreted. Literals include things like strings, Booleans, and integers.
```
Comment (binarylogic) [inline docs/reference/remap/concepts/literal.cue:5]:
```suggestion
		to be interpreted. Literals include things like strings, booleans, and integers.
```

**Review by binarylogic — APPROVED** [review]
Looks good.

File: docs/reference/remap/concepts/literal.cue (hunk 2 above)
Comment (binarylogic) [reply docs/reference/remap/concepts/literal.cue:5]: Didn't realize Boolean was a formal noun.

**Comment by lucperkins** [comment]
There are more sections I'd like to go over but I'll merge for now

**Pull request merged**